### ISSUE TITLE
Structure of log messages

### ISSUE THREAD

**Issue by antsr** [QudevETH/PycQED_py3#14]
We should come up with a common understanding, what we want to put in the different logging streams (`error`, `warning`, `info`, `debug`), and which streams we want printed in the jupyter notebook.

At the moment I can recall these types of messages that we use:
1. Printing out variable values for debugging purposes
2. Status updates of measurement progress
3. AWG upload status messages (this is way too verbose)
4. Analysis results
5. Warnings that parameters will not be updated
6. Other warnings (do we have any)
7. Errors? (do we have error messages that are not exceptions?)
8. Please add if you have encountered or would like to see some more types of messages.

My first suggestion would be that only MC and Pulsar status messages, in concise form, should go to the `info` stream. `error`, `warning` and `info` would be printed in the notebook by default. Variable values and analysis results would go to the debug stream, which can be enabled in a per-module basis (this should be easy to implement by adding and removing handlers).

Another open question is the formating of these log messages. I think we want to prepend the messages with the module name that the message comes from, atleast for error, warning and debug.

**Comment by nathlacroix** [comment]
I agree. To add a clarification which indicates for me the difference between a warning and an error log message:
- error message is used when an exception is raised and catched using and except clause, and the software knows how to handle it. Eg. : cannot set a parameter to a value when loading hdf5 file. Should display error message and then set the parameter to None (that's what is currently done).
- warning message is used to inform the user about some action he is doing and how this may result in unexpected behavior. Eg. not updating a parameter value because update is False.

**Comment by antsr** [comment]
To me the message about `update=False` should rather be info. Warnings should appear for abnormal operation, but update=False can be part of standard operation aswell.

**Comment by nathlacroix** [comment]
As a follow up: I noticed the following warning message was outputted when using flux pulses :
`WARNING:pycqed.measurement.waveform_control.pulsar:Element cphase_flux_el_segment_11 starts with non-zero entry on AWG2.
Element cphase_flux_el_segment_11 starts with non-zero entry on AWG2.`
Questions: 
- should this really be a warning ? I would see it more as a debug information
- one warning per segment is outputted, I think this really clutters the log.  Is that what we want?

**Comment by antsr** [comment]
@nathlacroix This should be a warning, because the physical output from the Tektronix AWG is the first sample of the next waveform when it is waiting for a trigger.
I think that one warning per element is useful in some cases to track down where the issue happens. We should rather fix this issue by modifying the waveform. For example by setting the first sample to zero "by hand".
